packages/dropdown/src/DropdownLabel/DropdownLabel.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useIdAllocator } from '@leafygreen-ui/hooks';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { useDropdownContext } from '../DropdownContext';
+
+import {
+  groupStyle,
+  labelStyle,
+  labelThemeStyle,
+} from './DropdownLabel.styles';
+import { DropdownLabelProps } from './DropdownLabel.types';
+
+export function DropdownLabel({
+  className,
+  label,
+  children,
+  ...rest
+}: DropdownLabelProps) {
+  const { theme } = useDarkMode();
+  const { renderedContext } = useDropdownContext();
+  const groupId = useIdAllocator({ prefix: 'select-option-group' });
+
+  return (
+    <div
+      className={cx(
+        groupStyle,
+        labelThemeStyle(theme, renderedContext),
+        className,
+      )}
+      {...rest}
+    >
+      <div id={groupId} className={cx(labelStyle)}>
+        {label}
+      </div>
+      <div role="group" aria-labelledby={groupId}>
+        {children}
+      </div>
+    </div>
```

Comment:
If this is how we expect to structure it, would it not be better to have an optional label prop on `DropdownGroup`?

Reply:
There's so much unshared logic I think it'd get messy :/